### PR TITLE
StateManager

### PR DESCRIPTION
**Pull request by white238**
This PR does two things:

* Breaks `StateManager` into it's own file
* Breaks a circular dependency between `infrastructure` and `physics/utilities` I mistakenly introduced

### CODE REVIEW

File: src/serac/infrastructure/output.hpp [hunk 1]
```diff
@@ -38,10 +38,11 @@ enum class Language
...
  * @param[in] datastore Root of the Sidre datastore
  * @param[in] file_name_prefix Prefix used for generating output file name
+ * @param[in] data_collection_name Name of the Data Collection stored in Sidre
  * @param[in] time Current simulation time
  * @param[in] language The output language format
  */
-void outputFields(const axom::sidre::DataStore& datastore, const std::string& file_name_prefix, double time,
-                  const Language language = Language::JSON);
+void outputFields(const axom::sidre::DataStore& datastore, const std::string& data_collection_name,
+                  const std::string& file_name_prefix, double time, const Language language = Language::JSON);
```
Comment (joshessman-llnl) [inline src/serac/infrastructure/output.hpp]: 5 params makes me a tiny bit nervous, especially the string parameters that are next to each other.  If we ever need to add further options possibly we could wrap all the parameters after the first one into a struct

File: src/serac/infrastructure/output.cpp [hunk 2]
```diff
@@ -50,12 +49,10 @@ void outputFields(const axom::sidre::DataStore& datastore, const std::string& fi
...
   // Get domain Sidre group
-  // TODO: get this from StateManager directly?
-  const axom::sidre::Group* sidre_root      = datastore.getRoot();
-  const std::string         collection_name = StateManager::collectionName();
-  SLIC_ERROR_ROOT_IF(!sidre_root->hasGroup(collection_name),
-                     fmt::format("Expected a datacollection root at '{0}' but it was not found", collection_name));
-  const axom::sidre::Group* domain_grp = sidre_root->getGroup(collection_name);
+  const axom::sidre::Group* sidre_root = datastore.getRoot();
+  SLIC_ERROR_ROOT_IF(!sidre_root->hasGroup(data_collection_name),
+                     fmt::format("Expected a datacollection root at '{0}' but it was not found", data_collection_name));
```
Comment (jamiebramwell) [inline src/serac/infrastructure/output.cpp]: Maybe adding `in sidre` or something to know what we are looking at?

**Review by joshessman-llnl — APPROVED** [review]

File: src/serac/physics/utilities/state_manager.cpp [hunk 3]
```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2019-2021, Lawrence Livermore National Security, LLC and
```
Comment (jamiebramwell) [inline src/serac/physics/utilities/state_manager.cpp:1]: Did you change anything in this other than moving the file?

File: src/serac/physics/utilities/state_manager.hpp [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+/**
+ * @file state_manager.hpp
+ *
+ * @brief This file contains the declaration of the StateManager class
+ */
+
+#pragma once
+
+//#include <functional>
```
Comment (jamiebramwell) [inline src/serac/physics/utilities/state_manager.hpp]: Dead code.

File: src/serac/infrastructure/output.cpp (hunk 2 above)
Comment (white238) [reply src/serac/infrastructure/output.cpp:54]: No, the only functional change was the breaking the circular dependancy.  This happened in output.cpp/hpp and in serac.cpp source files

File: src/serac/physics/utilities/state_manager.cpp (hunk 3 above)
Comment (white238) [reply src/serac/physics/utilities/state_manager.cpp:1]: No, the only functional change was the breaking the circular dependancy. This happened in output.cpp/hpp and in serac.cpp source files

**Review by jamiebramwell — APPROVED** [review]

File: src/serac/infrastructure/output.hpp (hunk 1 above)
Comment (white238) [reply src/serac/infrastructure/output.hpp]: I have an idea.. change incoming.

**Review by joshessman-llnl — APPROVED** [review]

**Pull request merged**